tuf/api/metadata.py

```diff
@@ -57,144 +55,94 @@ def __init__(self, signed: 'Signed', signatures: list) -> None:
 
         Returns:
             A TUF Metadata object.
 
         """
-        return cls.from_dict(load_json_file(filename, storage_backend))
+        if deserializer is None:
+            # Function-scope import to avoid circular dependency. Yucky!!!
+            # TODO: At least move to _get_default_metadata_deserializer helper.
+            from tuf.api.serialization.json import JSONDeserializer # pylint: disable=import-outside-toplevel
+            deserializer = JSONDeserializer()
 
+        if storage_backend is None:
+            storage_backend = FilesystemBackend()
 
-    # Serialization.
-    def to_dict(self) -> JsonDict:
-        """Returns the JSON-serializable dictionary representation of self. """
-        return {
-            'signatures': self.signatures,
-            'signed': self.signed.to_dict()
-        }
-
+        with storage_backend.get(filename) as file_obj:
+            raw_data = file_obj.read()
 
-    def to_json(self, compact: bool = False) -> None:
-        """Returns the optionally compacted JSON representation of self. """
-        return json.dumps(
-                self.to_dict(),
-                indent=(None if compact else 1),
-                separators=((',', ':') if compact else (',', ': ')),
-                sort_keys=True)
+        return deserializer.deserialize(raw_data)
 
 
-    def to_json_file(
-            self, filename: str, compact: bool = False,
-            storage_backend: StorageBackendInterface = None) -> None:
-        """Writes the JSON representation of self to file storage.
+    def to_file(self, filename: str, serializer: MetadataSerializer = None,
+                storage_backend: StorageBackendInterface = None) -> None:
+        """Writes TUF metadata to file storage.
 
         Arguments:
             filename: The path to write the file to.
-            compact: A boolean indicating if the JSON string should be compact
-                    by excluding whitespace.
+            serializer: A MetadataSerializer subclass instance that implements
+                the desired wireline format serialization. Per default a
+                JSONSerializer is used.
             storage_backend: An object that implements
                 securesystemslib.storage.StorageBackendInterface. Per default
                 a (local) FilesystemBackend is used.
+
         Raises:
+            tuf.api.serialization.SerializationError:
+                The metadata object cannot be serialized.
             securesystemslib.exceptions.StorageError:
                 The file cannot be written.
 
         """
+        if serializer is None:
+            # Function-scope import to avoid circular dependency. Yucky!!!
+            # TODO: At least move to a _get_default_metadata_serializer helper.
+            from tuf.api.serialization.json import JSONSerializer # pylint: disable=import-outside-toplevel
+            serializer = JSONSerializer(True) # Pass True to compact JSON
```

Comment:
```suggestion
            serializer = JSONSerializer(compact=True)
```
Using the named argument makes it clearer what's going on and obviates the need for the comment.